src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -47,49 +48,48 @@ public LetterService(LetterRepository letterRepository, Zipper zipper, ObjectMap
 
     @Transactional
     public UUID send(LetterRequest letter, String serviceName) {
-        Asserts.notEmpty(serviceName, "serviceName");
-
-        final String messageId = generateChecksum(letter);
-
-        log.info("Generated message: id = {}", messageId);
-
-        Optional<Letter> duplicateLetter = letterRepository.findByMessageIdAndStatusOrderByCreatedAtDesc(
-            messageId,
-            Created
+        return send(
+            new LetterInfo(generateChecksum(letter), serviceName, letter.type, letter.additionalData),
+            () -> pdfCreator.createFromTemplates(letter.documents)
         );
-
-        return duplicateLetter
-            .map(l -> {
-                UUID id = l.getId();
-                log.info("Same message found already created. Returning letter id {} instead", id);
-                return id;
-            })
-            .orElseGet(() -> saveNewLetterAndReturnId(letter, messageId, serviceName));
     }
 
     @Transactional
     public UUID send(LetterWithPdfsRequest letter, String serviceName) {
         throw new NotImplementedException();
     }
 
-    private UUID saveNewLetterAndReturnId(LetterRequest letterRequest, String messageId, String serviceName) {
+    private UUID send(LetterInfo letterInfo, Supplier<byte[]> pdfContentSupplier) {
+        Asserts.notEmpty(letterInfo.serviceName, "serviceName");
+
+        return letterRepository
+            .findByMessageIdAndStatusOrderByCreatedAtDesc(letterInfo.messageId, Created)
+            .map(duplicate -> {
+                UUID id = duplicate.getId();
+                log.info("Same message found already created. Returning letter id {} instead", id);
+                return id;
+            })
+            .orElseGet(() -> saveNewLetter(letterInfo, pdfContentSupplier.get()));
+    }
+
+    private UUID saveNewLetter(LetterInfo letterInfo, byte[] pdfContent) {
         UUID id = UUID.randomUUID();
 
         byte[] zipContent = zipper.zip(
             new PdfDoc(
-                FileNameHelper.generateName(letterRequest.type, serviceName, id, "pdf"),
-                pdfCreator.createFromTemplates(letterRequest.documents)
+                FileNameHelper.generateName(letterInfo.type, letterInfo.serviceName, id, "pdf"),
+                pdfContent
             )
         );
 
         // TODO: encrypt zip content
 
         Letter letter = new Letter(
             id,
-            messageId,
-            serviceName,
-            mapper.valueToTree(letterRequest.additionalData),
-            letterRequest.type,
+            letterInfo.messageId,
+            letterInfo.serviceName,
+            mapper.valueToTree(letterInfo.additionalData),
+            letterInfo.type,
             zipContent
         );
```

Comment:
Maybe to `LetterInfo` too? It is a helper class after all, should make use of it and simplify this crud operation